src/main/java/com/kobylynskyi/graphql/codegen/mapper/TypeDefinitionToDataModelMapper.java

```diff
@@ -80,22 +80,21 @@
      * @param document       Parent GraphQL document
      * @return Freemarker data model of the GraphQL type
      */
-    private static Set<ParameterDefinition> getFields(MappingConfig mappingConfig,
-                                                      ExtendedObjectTypeDefinition typeDefinition,
-                                                      ExtendedDocument document) {
-        // this includes parameters from base definition and extensions
-        List<ParameterDefinition> typeParameters = FieldDefinitionToParameterMapper.mapFields(mappingConfig,
-                typeDefinition.getFieldDefinitions(), typeDefinition.getName());
-        List<ParameterDefinition> typeParametersFromInterfaces = getInterfacesOfType(typeDefinition, document)
-                .stream()
+    private static Collection<ParameterDefinition> getFields(MappingConfig mappingConfig,
+                                                             ExtendedObjectTypeDefinition typeDefinition,
+                                                             ExtendedDocument document) {
+        // using the map to exclude duplicate fields from the type and interfaces
+        Map<String, ParameterDefinition> allParameters = new LinkedHashMap<>();
+
+        // includes parameters from the base definition and extensions
+        FieldDefinitionToParameterMapper.mapFields(mappingConfig, typeDefinition.getFieldDefinitions(), typeDefinition.getName())
+                .forEach(p -> allParameters.put(p.getName(), p));
+        // includes parameters from the interface
+        getInterfacesOfType(typeDefinition, document).stream()
```

Comment:
Why do we need to get fields from the interface in the first place?
By definition, for any implemented interface, all fields should be declared in the type itself as well.
Is it to get metadata?

Reply:
Sorry, I didn’t get your point.
We are getting all fields fields from the type first. And then adding fields from the interface

Reply:
Yes, that's why we need to read the interface's fields as well, indeed.
However, we could imagine a situation where the doc is overridden in the type (or not even present in the interface), and in that case we need to take the doc from the type's field, not the interface's one

Reply:
I've modified the logic to take JavaDoc and annotations from the type. 
And if something is absent then it will be taken from the interface field.

Reply:
Now I'm wondering, do we really want to have this at all? I mean, to generate the Java code for the interface, we use the interface fields and docs. And to generate the Java code for the implementing class, we don't need to look at the interface fields at all, do we?
Javadoc would naturally be inherited by overriding methods (we don't even need to do this at codegen level). And annotations are not inherited, but do we want that?

Reply:
Could you please create a separate issue with your suggestion? 
We can discuss it there.
Thanks!

Reply:
Sure I can do that, I just felt like it was another way to fix the issue at hand, not a separate issue.
But you're right, we can merge this and fix the bug right away.